File: src/main/java/threed/utils/ResourceUtils.java
package threed.utils;

import threed.common.ShaderLoader;

import java.io.*;

public class ResourceUtils {
        
    public static String loadStringResource(String path, String encoding) throws IOException {
        InputStream in = ShaderLoader.class.getClassLoader().getResourceAsStream(path);
        if (in == null)
            throw new FileNotFoundException("no such file: " + path);

        try {
            Reader reader = new InputStreamReader(in, encoding);
            char[] buffer = new char[1024];
            int n;
            StringBuilder sb = new StringBuilder();

            while ((n = reader.read(buffer)) != -1)
                sb.append(buffer, 0, n);

            return sb.toString();
            
        } finally {
            in.close();
        }
    }
}

File: src/main/java/threed/utils/ResourceUtils.kt
package threed.utils

import java.io.*

object ResourceUtils {
        
    fun loadStringResource(path: String, encoding: String): String {
        val input = javaClass.getClassLoader()!!.getResourceAsStream(path)
        if (input != null)
            return input.use { it.reader(encoding).readText() }
        else
            throw FileNotFoundException("no such file: $path")
    }
}
